azblob/zc_retry_reader.go

```diff
@@ -120,3 +140,57 @@ func (s *retryReader) Close() error {
 	}
 	return nil
 }
+
+// Returns a function that can be used to force a retry within (and transparently to) an call to Read
+func (s *retryReader) getForceRetryFuncOrNil() func(){
```

Comment:
This is very specific to the use case in AzCopy, I'm debating whether this should be kept out of the SDK.

Reply:
In hindsight, I think the signature should have been 
```func (s *retryReader) getForceRetryFunc() (func(), error) { ```

That's more idomatic in terms of the error handling.  Do you think that would make it more suitable for inclusion in the SDK?  If not, what would be a good way to take it out of the SDK?

Oh, BTW, the reason I made the method private was to avoid cluttering up the SDK too much. Being private, its only exposed indirectly in the SDKs public API, in the ```BodyWithForceableRetry``` method in response_helpers.

Reply:
With recent refactoring, this routine is no longer required